### PR TITLE
Fact branch

### PR DESCRIPTION
**Pull request by Shailja2606**
adding programs with class function

### CODE REVIEW

**Comment by Shailja2606** [comment]
updated factorial and palindrom program with class

File: shailja_work/calculator_class.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        return num1 - num2
+
+    def multiply(self, num1, num2):
+        return num1 * num2
+
+    def divide(self, num1, num2):
+        return num1 / num2
+
+
+obj_cal = Calculator()
```
Comment (Patibandha) [inline shailja_work/calculator_class.py]: need to call object of class inside if your main() method line #20

File: shailja_work/factorial_class.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        if f < 0:
+            msg = "Please Enter Positive number"
+            return msg
+        elif f == 0 or f == 1:
+            return 1
+        else:
+            return f * factorial(f - 1)
+
+
+obj_fac = Fact()
```
Comment (Patibandha) [inline shailja_work/factorial_class.py:17]: same mistake it suppose to from your main() function line #20

File: shailja_work/palindrom_class.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+    def palstr(self, rem_spe_chr):
+        if rem_spe_chr == rem_spe_chr[::-1]:
+            msg = "The string is a palindrome"
+        else:
+            msg = "Not a palindrome"
+        logging.warning(msg)
+
+
+obj_pali = Pali()
```
Comment (Patibandha) [inline shailja_work/palindrom_class.py:24]: this is wrong!

File: shailja_work/test_calculator_class.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import pytest
+
+from PD008bootcamp.shailja_work.calculator_class import obj_cal
+
+
```
Comment (Patibandha) [inline shailja_work/test_calculator_class.py:12]: it suppose to be -ve test cases

File: shailja_work/test_factorial_class.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import unittest
+
+from PD008bootcamp.shailja_work.factorial_class import obj_fac
+
```
Comment (Patibandha) [inline shailja_work/test_factorial_class.py:4]: more -ve test cases needed

File: shailja_work/test_palindrome_class.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import logging
+
+from PD008bootcamp.shailja_work.palindrom_class import obj_pali
+
+LOGGER = logging.getLogger(__name__)
+
+
+def test_pal_true():
+    LOGGER.warning("The string is a palindrome")
```
Comment (Patibandha) [inline shailja_work/test_palindrome_class.py:9]: what about assert false

**Review by Patibandha — CHANGES_REQUESTED** [review]
made changes

**Pull request merged**